static/app/components/stacktracePreview.tsx

```diff
@@ -22,220 +22,208 @@ import getThreadStacktrace from './events/interfaces/threads/threadSelector/getT
+      if (loaderTimeout.current !== null) {
+        window.clearTimeout(loaderTimeout.current);
+      }
+      if (delayTimeout.current !== null) {
+        window.clearTimeout(delayTimeout.current);
+      }
+    };
+  }, []);
+
+  const fetchData = React.useCallback(async () => {
+    // Data is already loaded
+    if (event) {
+      return;
     }
-  };
-
-  fetchData = async () => {
-    const {organization, api, issueId, eventId, projectSlug} = this.props;
 
-    if (this.state.event || (!issueId && !(eventId && projectSlug))) {
+    // These are required props to load data
+    if (!props.issueId && !props.eventId && !props.projectSlug) {
       return;
     }
 
-    this.loaderTimeout = window.setTimeout(() => {
-      this.setState({loadingVisible: true});
+    loaderTimeout.current = window.setTimeout(() => {
+      setLoadingVisible(true);
     }, HOVERCARD_DELAY);
 
     try {
-      const event = await api.requestPromise(
-        eventId && projectSlug
-          ? `/projects/${organization.slug}/${projectSlug}/events/${eventId}/`
-          : `/issues/${issueId}/events/latest/?collapse=stacktraceOnly`
+      const evt = await props.api.requestPromise(
+        props.eventId && props.projectSlug
+          ? `/projects/${props.organization.slug}/${props.projectSlug}/events/${props.eventId}/`
+          : `/issues/${props.issueId}/events/latest/?collapse=stacktraceOnly`
       );
-      clearTimeout(this.loaderTimeout);
-      this.setState({event, loading: false, loadingVisible: false});
+      clearTimeout(loaderTimeout.current);
+      setEvent(evt);
+      setStatus('loaded');
+      setLoadingVisible(false);
     } catch {
-      clearTimeout(this.loaderTimeout);
-      this.setState({loading: false, loadingVisible: false});
-    }
-  };
-
-  handleStacktracePreviewClick = (event: React.MouseEvent) => {
-    event.stopPropagation();
-  };
-
-  getStacktrace(): StacktraceType | undefined {
-    const {event} = this.state;
-
-    if (!event) {
-      return undefined;
-    }
-
-    const exceptionsWithStacktrace =
-      event.entries
-        .find(e => e.type === EntryType.EXCEPTION)
-        ?.data?.values.filter(({stacktrace}) => defined(stacktrace)) ?? [];
-
-    const exceptionStacktrace: StacktraceType | undefined = isStacktraceNewestFirst()
-      ? exceptionsWithStacktrace[exceptionsWithStacktrace.length - 1]?.stacktrace
-      : exceptionsWithStacktrace[0]?.stacktrace;
-
-    if (exceptionStacktrace) {
-      return exceptionStacktrace;
-    }
-
-    const threads =
-      event.entries.find(e => e.type === EntryType.THREADS)?.data?.values ?? [];
-    const bestThread = findBestThread(threads);
-
-    if (!bestThread) {
-      return undefined;
+      clearTimeout(loaderTimeout.current);
+      setEvent(null);
+      setStatus('error');
+      setLoadingVisible(false);
     }
+  }, []);
 
-    const bestThreadStacktrace = getThreadStacktrace(false, bestThread);
+  const handleMouseEnter = React.useCallback(() => {
```

Comment:
Could this timeout be removed in favor of `displayTimeout` from Hovercard? That way we could have a delay on displaying the card, but then immediately load the data for the component and avoid all the timeout code

Reply:
we were trying to reduce the number of api calls if someone happens to scroll the issue stream while hovering the area that triggers the stacktrace because these api calls include calls to snuba adding additional load. 

but we later went back and reduced the number of snuba calls via the "stacktraceOnly" query parameter. It might still include one snuba query?

Reply:
@scttcper Ah, I see. I think we could delay the hovercard opening in this case and just fetch all the data then, this would only trigger the api call if elapsed > hovercard delay prop. It's slightly different UX, but idk.